### DISCUSSION TITLE
vi/models/sam/

### DISCUSSION THREAD

**Discussion by giscus** [ultralytics/ultralytics#19050] [Docs]
# vi/models/sam/

Khám phá Mô hình Segment Anything mang tính cách mạng ( SAM ) để phân đoạn hình ảnh nhanh chóng với hiệu suất không cần chụp. Khám phá các tính năng chính, tập dữ liệu và mẹo sử dụng.

https://docs.ultralytics.com/vi/models/sam/

**Comment by UltralyticsAssistant** [comment]
👋 Hello, thank you for your interest in Ultralytics 🚀! We recommend a visit to the [Docs](https://docs.ultralytics.com/) where you can explore the revolutionary [Segment Anything Model (SAM)](https://docs.ultralytics.com/vi/models/sam/) for efficient image segmentation without the need for fine-tuning. The page details key features, datasets, and practical usage tips to help you get started!

If this is a 🐛 Bug Report, please provide a [minimum reproducible example](https://docs.ultralytics.com/help/minimum-reproducible-example/) to facilitate debugging.

If this is a custom usage ❓ Question related to SAM, please provide as much information as possible, including example inputs, expected outcomes, and details about your environment to better assist you.

Join the Ultralytics community to connect with others just like you! For real-time discussions, visit us on [Discord](https://discord.com/invite/ultralytics) 🎧. Prefer forum-style conversations? Head over to [Discourse](https://community.ultralytics.com/). Or find exciting discussions on our [Subreddit](https://reddit.com/r/Ultralytics).

## Upgrade

Make sure you're using the latest `ultralytics` package, which includes all [requirements](https://github.com/ultralytics/ultralytics/blob/main/pyproject.toml), in a supported [**Python>=3.8**](https://www.python.org/) environment with [**PyTorch>=1.8**](https://pytorch.org/get-started/locally/) installed:

```bash
pip install -U ultralytics
```

## Environments

You can experiment with SAM in the following verified environments, preloaded with all dependencies like [Python](https://www.python.org/) and [PyTorch](https://pytorch.org/):

- **Notebooks** with free GPU: <a href="https://console.paperspace.com/github/ultralytics/ultralytics"><img src="https://assets.paperspace.io/img/gradient-badge.svg" alt="Run on Gradient"/></a> <a href="https://colab.research.google.com/github/ultralytics/ultralytics/blob/main/examples/tutorial.ipynb"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"></a> <a href="https://www.kaggle.com/models/ultralytics/yolo11"><img src="https://kaggle.com/static/images/open-in-kaggle.svg" alt="Open In Kaggle"></a>
- **Google Cloud** Deep Learning VM: See the [GCP Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/google_cloud_quickstart_tutorial/)
- **Amazon** Deep Learning AMI: See the [AWS Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/aws_quickstart_tutorial/)
- **Docker Image**: See the [Docker Quickstart Guide](https://docs.ultralytics.com/yolov5/environments/docker_image_quickstart_tutorial/) <a href="https://hub.docker.com/r/ultralytics/ultralytics"><img src="https://img.shields.io/docker/pulls/ultralytics/ultralytics?logo=docker" alt="Docker Pulls"></a>

## Status

<a href="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule"><img src="https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml/badge.svg" alt="Ultralytics CI"></a>

If this badge is green, all [Ultralytics CI](https://github.com/ultralytics/ultralytics/actions/workflows/ci.yml?query=event%3Aschedule) tests are currently passing. CI tests ensure proper functioning of all YOLO and SAM [Modes](https://docs.ultralytics.com/modes/) and [Tasks](https://docs.ultralytics.com/tasks/) on macOS, Windows, and Ubuntu every 24 hours and whenever updates are committed.

This is an automated response to assist you quickly 😊. An Ultralytics engineer will follow up with more details soon!
